### DISCUSSION TITLE
Equivalent of toctree hidden?

### DISCUSSION THREAD

**Discussion by matthew-brett** [executablebooks/meta#105] [Q&A] [answered]
I've got quite a few notebooks that I don't have in my `_toc.yml`, deliberately.

These do build, of course, but I get lots of warnings in my log output.   Is there a way of telling the build that I meant to leave those files out of the toctree - the equivalent of the Sphinx `toctree: hidden` construct?

**Comment by chrisjsewell** [comment]
Yes you can use the [sphinx orphan](https://www.sphinx-doc.org/en/master/usage/restructuredtext/field-lists.html?highlight=orphan#special-metadata-fields) metadata, i.e. add `{"orphan": true}` to the notebook's metadata.
This is used, for example, for this page: https://myst-nb.readthedocs.io/en/latest/use/orphaned_nb.html#orphaned-nb
Let me know if that works for you 😄

**Comment by matthew-brett** [comment, marked as answer]
Excellent - yes it does, thanks.

Should there be a top-level `jupyterbook` metadata namespace for things like this, do you think?

**Reply by chrisjsewell** [reply]
ass in you list all these orphaned notebooks somewhere in `_config.yml`, then inject `orphan=true` in to the notebooks during the build (if not already present)?

That would be possible

**Reply by matthew-brett** [reply]
No - sorry - I'm happy to drop `orphan: true` into the metadata of the non-toc notebook - I was just wondering about having that metadata at the notebook metadata top level - being more likely to cause clashes with some other user of metadata.   As in maybe it would be safer to have (YaML version):

```
jupyter:
  jupyterbook:
    orphan: true
```

instead of 

```
jupyter:
  orphan: true
```

**Reply by chrisjsewell** [reply]
Err, this would make it less trivial to convert the notebook metadata to sphinx metadata, i.e. you would have to start adding custom logic to translate key paths. So its not something that I would consider, unless a lot of people start asking for it

**Comment by matthew-brett** [comment]
I'm sure there's something I don't understand about the translation.   Would it not just be a case of translatiing only the `jupyterbook` sub-key metadata, rather than all the `jupyter` metadata?  Or merging the `jupyterbook` key into the top-level metadata, just for the translation? I guess there must be other top-level jupyter metadata that the translation uses ...

**Comment by matthew-brett** [comment]
Actually - I spoke too soon - or something is changing that I don't understand.  I can't get `orphan: true` to work.

Latest master of my example repo includes your https://myst-nb.readthedocs.io/en/latest/use/orphaned_nb.html#orphaned-nb file, but still generates a warning in the build step about the file not being in the _toc file.

Example repo: https://github.com/matthew-brett/jbtest

**Reply by matthew-brett** [reply]
Do you get the same not-supressed warning as me?  Should I put this as an issue in the tracker?
